### ISSUE TITLE
Make the object rotating in the same position

### ISSUE THREAD

**Issue by mecherguikhaoula1234** [SceneView/sceneview-android#352]
Hello ,
I want to make my object rotating in the same position without moving in the whole surface in a SceneView . I tried many functions to fix the position of camera to be the same of the object and i used transform but it didn't work . 
You can find my code to add a 3D object in SceneView below and what i tried in comments to fix the position of the object and camera and rotate the object around itself.

```
       val modelAsset = modelAssets[value]
                   var modelInstance3D =  sceneView_3D.modelLoader.loadModelInstance(modelAsset.fileLocation)!!
             
                   modelNode = ModelNode(
                        modelInstance3D,
                        scaleToUnits = 2f,
                        centerOrigin = Position(0f, 0f, 0f),
                        autoAnimate = true
                    ).apply {

                       /* transform(
                            position = Position(0f, 0f, 0f), // Set the position to (0, 0, 0) to keep it fixed
                            rotation = Rotation(0f, 0f, 45f) // Adjust the rotation as needed (45 degrees in this example)
                        )*/
                     /* transform(
                            position = Position(
                                0f
                            ),
                            rotation = Rotation()
                        )*/
                        // isEditable =  true
                       
                  }
                    //   sceneView_3D.cameraNode.lookTowards(Direction(0f,0f,0f))
                    //   sceneView_3D.cameraNode.lookAt(modelNode!!.position)
                    //  sceneView_3D.cameraNode.camera.lookAt(Position(0f, 0f,0f),Position(0f, 0f,0f),Position(0f, 0f,0f))
                  //  sceneView_3D.cameraNode.lookAt(modelNode!!)
                   // sceneView_3D.cameraNode.worldToScreenPoint(Vector3(0f, 0f,0f))
                    //sceneView_3D.cameraNode.rotation =  Rotation(x = 0.0f, y = 0.0f, z = 0.0f)
                  //  sceneView.cameraNode.position = Position(x = 0.0f, y = 0.0f, z = 0.0f)

                    sceneView_3D.addChildNode(modelNode!!)
```

Thank you in advance

**Comment by marmpa** [comment]
Ok i don't know if this works for your case, since it was created for a version on August 2023, but i did that exact thing using this code:
`onUpdate = {node, _->


                    //arvideoNode. node.worldPosition
                    //Toast.makeText(applicationContext, "NAII2",Toast.LENGTH_SHORT).show()

                    if(node.isTracking){
                        Toast.makeText(applicationContext, "TRACKING: "+arvideoNode.player.isPlaying.toString(),Toast.LENGTH_SHORT).show()

                        val imagePose = node.pose

                        arvideoNode.worldPosition = Position(imagePose!!.tx(), imagePose!!.ty(), imagePose!!.tz())

                        val currentQuaternion = Quaternion(imagePose.qx(), imagePose.qy(), imagePose.qz(), imagePose.qw())

                        // Create a quaternion that represents a 180 degrees rotation around the x-axis
                        val rotationQuaternion = Quaternion.fromAxisAngle(Float3(1.0f, 0.0f, 0.0f), 180.0f)

                        // Combine the current orientation with the rotation

                        arvideoNode.worldQuaternion = Quaternion(imagePose.qx(), imagePose.qy(), imagePose.qz(), imagePose.qw())
                        arvideoNode.worldQuaternion = currentQuaternion * rotationQuaternion
                        //arvideoNode.worldQuaternion.fromAxisAngle()
                        //arvideoNode.wo
                        if(!arvideoNode.player.isPlaying)
                        {
                            //arvideoNode.quaternion= node.augmentedImage!!.

                            arvideoNode.player.start()
                        }
                        else{
                            if(arvideoNode.player.isPlaying)
                            {
                                arvideoNode.player.pause()
                            }
                        }
                    }
                }`
                
Basically, I get the current pose(which i guess is the direction the object is detected by the AR gods), and then i rotate it based on the Quaternion. Don't ask me for too much, i haven't figured out exactly all the math of how it works, i just get the context of it.

**Comment by mecherguikhaoula1234** [comment]
Hello
Thank you for your return but i found a simpler solution   by calculating the angle of rotation in the touchlistener of the component(you can also zoom in and zoom out the modelnode) so i post the code below: 

  ```
  var rotationAngleX =0f
    var rotationAngleY =0f
    var previousX = 0f
    var previousY = 0f
    var sensitivity = 0.5f
    var zoomFactor = 1.0f
    var initialDistance = 0.0f
    var minZoom = 0.5f
    var maxZoom =3f

sceneView_3D.setOnTouchListener { v, event ->
            if (modelNode != null) {

                val touchX = event.x
                val touchY = event.y

                when (event.action) {
                    MotionEvent.ACTION_POINTER_DOWN -> {
                        if (event.pointerCount >= 2) {
                            initialDistance = calculateFingerSpacing(event)
                        }
                    }
                    MotionEvent.ACTION_MOVE -> {
                        if (event.pointerCount >= 2) {
                            val newDistance: Float = calculateFingerSpacing(event)
                            if (newDistance > 0) {
                                zoomFactor *= newDistance / initialDistance
                                // Limit the zoom range if needed
                                zoomFactor = Math.max(minZoom, Math.min(maxZoom, zoomFactor))
                                //zoomFactor = Math.max(modelNode!!.minEditableScale, Math.min(maxZoom, modelNode!!.maxEditableScale))
                                initialDistance = newDistance
                                 modelNode!!.scale = Scale(zoomFactor)
                            }
                        } else {
                            val dx: Float = (touchX - previousX) * sensitivity
                            val dy: Float = (touchY - previousY) * sensitivity
                            rotationAngleX += dy
                            rotationAngleY += dx
                        }
                    }
                }

                previousX = touchX
                previousY = touchY
                modelNode!!.rotation = Rotation(rotationAngleX, rotationAngleY)
            }
            return@setOnTouchListener true
        }
```

knowing that sceneview_3D is a  **<io.github.sceneview.SceneView**  and i give you example of initialization of modelNode:  

 ```
 modelInstance = ModelLoader(sceneView_3D.engine, this@AR3DActivity).loadModel(modelAsset.fileLocation)!!.instance
  modelNode = ModelNode(
                    modelInstance!!,
                    scaleToUnits = 0.5f,
                    centerOrigin = Position(0f, 0f, 0f),
                    autoAnimate = true
                ).apply {
                    //transform(position = Position(0f, 0f, 1f), rotation = Rotation())
                    isEditable = true
                }
               sceneView_3D.addChildNode(modelNode!!)


```
